api/envoy/config/filter/network/thrift_proxy/v2alpha1/route.proto

```diff
@@ -43,9 +44,19 @@ message RouteMatch {
     string service_name = 2;
   }
 
-  // Inverts whatever matching is done in match_specifier. Cannot be combined with wildcard matching
-  // as that would result in routes never being matched.
+  // Inverts whatever matching is done in the :ref:`method_name
```

Comment:
Does this also invert the matching done in the `headers`?

Reply:
it does not -- I explicitly mention the `method_name` and `service_name` fields in the hopes of making it clear that this doesn't apply to headers(originally to address @zuercher's same concern: https://github.com/envoyproxy/envoy/pull/4239#discussion_r212486618). 

I can certainly add language to make it explicit that it doesn't apply to header matching if you think that'll help.

Reply:
Got it, thanks. I think would help to add a bit of language for that, just to err on the side of caution.